src/consensus/aft/raft_types.h

```diff
@@ -92,6 +93,16 @@ namespace aft
       }
       return nullptr;
     }
+
+    virtual bool record_committable(Index idx) override
```

Comment:
This isn't necessary. Since the `kv::Store` has now access to the snapshotter directly, it can record a committable index when it `commit`s a `globally_committable` entry.

Reply:
There are two calls (1 leader, 1 backup) to `record_committable` left in `raft.h` and both of them are used to determine whether a ledger chunk hast to be forced, i.e. this goes into `ledger->put_entry`, so the ledger would have to have a reference to the store or snapshotter. I think I know how to do this from the store before getting into raft, but I might break some stuff before it starts working again :)